filters/valve.cpp

```diff
@@ -0,0 +1,205 @@
+#include <atomic>
+#include <string>
+
+#include "glog/logging.h"
+
+#include "process/help.hpp"
+#include "process/http.hpp"
+#include "process/future.hpp"
+#include "process/limiter.hpp"
+#include "process/process.hpp"
+
+#include "stout/lambda.hpp"
+#include "stout/synchronized.hpp"
+
+#include "valve.hpp"
+
+namespace mesos {
+namespace serenity {
+
+// TODO(bplotka): Break into explicit using-declarations.
+using namespace process;  // NOLINT(build/namespaces)
+
+using std::string;
+using std::atomic_bool;
+
+static const string ESTIMATOR_VALVE_ENDPOINT_HELP() {
+  return HELP(
+      TLDR(
+          "Disable or enable Serenity Resource Estimator."),
+      USAGE(
+          VALVE_ROUTE),
+      DESCRIPTION(
+          "This endpoint is a valve for Serenity Resource Estimator. ",
+          "When turned off it stops estimating new slack resources. ",
+          "",
+          "The following field should be supplied in a POST:",
+          "1. " + PIPELINE_ENABLE_KEY + " - true / false."));
+}
+
+
+static const string CONTROLLER_VALVE_ENDPOINT_HELP() {
+  return HELP(
+      TLDR(
+          "Disable or enable Serenity QoS Controller."),
+      USAGE(
+          VALVE_ROUTE),
+      DESCRIPTION(
+          "This endpoint is a valve for QoS Controller. ",
+          "WARINIG: When turned off it stops assuring QoS for ",
+          "Oversubscription feature - Best Effort executors can possibly",
+          "impact Production executors. ",
+          "Otherwise QoS Controller will assure QoS, by preemption of BE ",
+          "tasks. ",
+          "",
+          "The following field should be supplied in a POST:",
+          "1. " + PIPELINE_ENABLE_KEY + " - true / false."));
+}
+
+
+Try<string> getFormValue(
+    const string& key,
+    const hashmap<string, string>& values) {
+  Option<string> value = values.get(key);
+
+  if (value.isNone()) {
+    return Error("Missing value for '" + key + "'.");
+  }
+
+  // HTTP decode the value.
+  Try<string> decodedValue = process::http::decode(value.get());
+  if (decodedValue.isError()) {
+    return decodedValue;
+  }
+
+  // Treat empty string as an error.
+  if (decodedValue.isSome() && decodedValue.get().empty()) {
+    return Error("Empty string for '" + key + "'.");
+  }
+
+  return decodedValue.get();
+}
+
+
+class ValveFilterEndpointProcess : public Process<ValveFilterEndpointProcess> {
+ public:
+  ValveFilterEndpointProcess(ValveType _valveType, bool _opened)
+    : ProcessBase(getProcessBaseName(_valveType)),
+      valveType(_valveType),
+      opened(_opened),
+      limiter(2, Seconds(1)) {}  // 2 permits per second.
+
+  virtual ~ValveFilterEndpointProcess() {}
+
+  void setOpen(bool open) {
+    // NOTE: In future we may want to trigger some actions here.
+    this->opened = open;
```

Comment:
Let's emit a log message here.  To disambiguate the calls, perhaps it would be good to add some identifier to the filter.